### PR TITLE
[Defender (RiskIQ) - EASM] Api version 2023-05-01-preview

### LINKED ISSUE #24466: [RiskIQ - EASM - Defender EASM] API Review
New API Review meeting has been requested.

**Service Name**: RiskIQ - EASM - Defender EASM
**Review Created By**: Philip Wood @mspwood
**Review Date**: 07/18/2023 01:00 PM PT
**Onboarding Record**: https://dev.azure.com/azure-sdk/Release/_workitems/edit/14798
**PR**: https://github.com/Azure/azure-rest-api-specs/pull/24465
**Hero Scenarios Link**: [here](https://learn.microsoft.com/en-us/azure/external-attack-surface-management/)
**Core Concepts Doc Link**: [here](https://learn.microsoft.com/en-us/azure/external-attack-surface-management/)

**Description**: - Add a new Microsoft.Easm Dataplane Version 2023-05-01-preview.
- Support exporting assets for a provided filter.
- Support a unified observations endpoint for paginated retrieval of observations related to an asset.

Detailed meeting information and documents provided can be accessed [here](https://web.powerapps.com/apps/e41c69de-ef96-44ba-b287-0e012e7c13ae?view=edit&event=e20751a3-980b-ee11-8f6e-000d3a341bd5)
For more information that will help prepare you for this review, the requirements, and office hours, visit the documentation [here](https://aka.ms/azsdk/onboarding/restapischedule)

### PR DESCRIPTION
**Pull request by mspwood**
# Data Plane API - Pull Request
- Add a new Microsoft.Easm Dataplane Version 2023-05-01-preview.
- Support exporting assets for a provided filter.
- Support a unified observations endpoint for paginated retrieval of observations related to an asset.

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*API Spec engagement record*](https://aka.ms/azsdk/onboarding/restapischedule).

* Link to API Spec engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [x] a public preview
- [ ] GA release

### Change Scope
<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous API Spec document (if applicable), and the root paths that have been updated. </sup>
* Design Document: 
    * Observation Unification Design: https://microsoft.sharepoint.com/:w:/t/RiskIQ-Product/ESDNVwTKC5JKuK4gepDedPMByAwHiL46BbwcJu20NzLvbQ?e=ONtyAz
* Previous API Spec Doc:  https://github.com/Azure/azure-rest-api-specs/blob/main/specification/riskiq/data-plane/Microsoft.Easm/preview/2023-03-01-preview/easm.json
* Updated paths:
    * Observations - Added:  /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/workspaces/{workspaceName}/assets/{assetId}/observations
    * Asset Export - Added: 
 /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/workspaces/{workspaceName}/assets:export

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azsdk/onboarding/restapischedule) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azsdk/onboarding/restapischedule).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling
 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://aka.ms/style)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications
 * [Azure REST API Guidelines](https://aka.ms/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links
 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

fix https://github.com/Azure/azure-rest-api-specs/issues/24466

### CODE REVIEW

**Comment by weidongxu-microsoft** [comment]
Did we have a review on it?

**Comment by mikekistler** [comment]
Reopen for REST API review on 7/18/23

File: specification/riskiq/data-plane/Microsoft.Easm/preview/2023-05-01-preview/easm.json [hunk 1]
```diff
@@ -0,0 +1,5761 @@
...
+                "type": "string"
+              }
+            },
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "tags": ["Assets"],
+        "operationId": "Assets_Observations",
```
Comment (mikekistler) [inline specification/riskiq/data-plane/Microsoft.Easm/preview/2023-05-01-preview/easm.json:241]: The operationId for get on a collection should contain "list"

File: specification/riskiq/data-plane/Microsoft.Easm/preview/2023-05-01-preview/easm.json [hunk 2]
```diff
@@ -0,0 +1,5761 @@
...
+          },
+          {
+            "$ref": "#/parameters/FilterParameter"
+          },
+          {
+            "in": "query",
+            "name": "fileName",
+            "required": false,
+            "type": "string"
+          },
```
Comment (mikekistler) [inline specification/riskiq/data-plane/Microsoft.Easm/preview/2023-05-01-preview/easm.json:275]: All parameters should have a description.

File: specification/riskiq/data-plane/Microsoft.Easm/preview/2023-05-01-preview/easm.json [hunk 3]
```diff
@@ -0,0 +1,5761 @@
...
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "tags": ["Reports"],
+        "operationId": "Reports_Snapshot_Export",
+        "summary": "Queue an export of the most recent snapshot of asset summary values to a file.",
+        "x-ms-examples": {
+          "Reports": {
+            "$ref": "./examples/Reports_Snapshot_Export.json"
```
Comment (mikekistler) [inline specification/riskiq/data-plane/Microsoft.Easm/preview/2023-05-01-preview/easm.json:1184]: invalid-ref error ENOENT: no such file or directory

File: specification/riskiq/data-plane/Microsoft.Easm/preview/2023-05-01-preview/easm.json [hunk 4]
```diff
@@ -0,0 +1,5761 @@
...
+          "type": "integer"
+        },
+        "value": {
+          "description": "The items in the current page of results.",
+          "items": {
+            "$ref": "#/definitions/AssetResponse"
+          },
+          "type": "array"
+        }
+      },
```
Comment (mikekistler) [inline specification/riskiq/data-plane/Microsoft.Easm/preview/2023-05-01-preview/easm.json:1867]: The `value` property in pageable response should be required. This error occurs several more times ... please fix all.

File: specification/riskiq/data-plane/Microsoft.Easm/preview/2023-05-01-preview/easm.json [hunk 5]
```diff
@@ -0,0 +1,5761 @@
...
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/WorkspaceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/DiscoGroupNameParameter"
```
Comment (mikekistler) [inline specification/riskiq/data-plane/Microsoft.Easm/preview/2023-05-01-preview/easm.json:885]: Parameter "groupName" must be used in path.

**Review by mikekistler — CHANGES_REQUESTED** [review]
I flagged a number of issues reported by the Spectral linter. You could check for these yourself -- just follow the instructions at

https://github.com/Azure/azure-api-style-guide/blob/main/README.md#how-to-use-the-spectral-ruleset

**Pull request closed without merging**